tools/shared.py

```diff
@@ -884,9 +884,12 @@ def get_llvm_target():
                                  '-D__EMSCRIPTEN_tiny__=' + str(EMSCRIPTEN_VERSION_TINY)]
 
 if LLVM_TARGET == WASM_TARGET:
-  # wasm target does not automatically define emscripten stuff, so do it here
+  # wasm target does not automatically define emscripten stuff, so do it here.
+  # Also, set the ABI version to at least 2 so that we get nicely aligned string
+  # data and other nice fixes.
   COMPILER_OPTS = COMPILER_OPTS + ['-DEMSCRIPTEN',
-                                   '-D__EMSCRIPTEN__']
+                                   '-D__EMSCRIPTEN__',
+                                   '-D_LIBCPP_ABI_VERSION=2']
```

Comment:
Also fine for me. We have stressed to developers that .bc files are not compatible across Emscripten versions and that there is no locked in ABI across versions, so we should be free to do nimble stuff like this, where it helps. It is very good to note known breakages however, since these are good type of changes to document then in ChangeLog.markdown when I collect the tag updates there.